flash/data/data_pipeline.py

```diff
@@ -0,0 +1,224 @@
+                f'{self.__class__.__name__}: post_collate and gpu_pre_collate are mutual exclusive.'
+            )
+
+        elif post_collate_overriden:
+            worker_collate = collate_fn
+            device_collate = self._do_nothing_collate
+
+        elif device_pre_collate_overriden:
+            worker_collate = self._do_nothing_collate
+            device_collate = collate_fn
+
+        else:
+            worker_collate = collate_fn
+            device_collate = self._do_nothing_collate
+
+        worker_callable = Collater(worker_collate, self.pre_collate, self.post_collate)
+        device_callable = Collater(device_collate, self.device_pre_collate, self.device_post_collate)
+
+        return worker_callable, device_callable
+
+    @staticmethod
+    def _model_transfer_to_device_wrapper(func: Callable, collater: Collater) -> Callable:
+
+        @wraps(func)
+        def new_func(*args, **kwargs):
+            moved_to_device = func(*args, **kwargs)
+            return collater(moved_to_device)
+
+        return new_func
+
+    def _attach_to_model(self, model: LightningModule, loader_stage: str = 'all') -> LightningModule:
+        if loader_stage == 'all':
+            loader_stage = ['train', 'test', 'val', 'predict']
+
+        elif isinstance(loader_stage, str):
+            loader_stage = [loader_stage]
+
+        for stage in loader_stage:
+            loader_name = f'{stage}_loader'
+
+            if hasattr(model, loader_name):
+                dataloader = getattr(model, loader_name)
+
+                if isinstance(dataloader, _PatchDataLoader):
+                    wrap_patch_loader = True
+                    dataloader = dataloader()
+
+                else:
+                    wrap_patch_loader = False
+
+                if isinstance(dataloader, Sequence):
+                    was_seq = True
+                else:
+                    dataloader = [dataloader]
+                    was_seq = False
+
+                for idx, loader in enumerate(dataloader):
+                    if isinstance(loader, DataLoader):
+                        dl_args = {k: v for k, v in vars(loader).items() if not k.startswith("_")}
+
+                        dl_args['collate_fn'], device_collater = self.split_around_collate(
+                            collate_fn=dl_args['collate_fn']
+                        )
+
+                        loader = type(loader)(**dl_args)
+
+                    dataloader[idx] = loader
+
+                if not was_seq:
+                    dataloader = dataloader[0]
+
+                if wrap_patch_loader:
+                    dataloader = _PatchDataLoader(dataloader)
+
+                setattr(model, loader_name, dataloader)
+
+        model.transfer_batch_to_device = (
+            self._model_transfer_to_device_wrapper(model.transfer_batch_to_device, device_collater)
+        )
+        return model
+
+    def _generate_auto_dset(self, data: Union[Iterable, Any]) -> AutoDataset:
```

Comment:
should me make this public API? @tchaton

Reply:
I think `to_dataloader ` is enough.